File: backend/src/main/java/com/oose/labsafety/laboratory/domain/Laboratory.java
package com.oose.labsafety.laboratory.domain;

import com.oose.labsafety.common.domain.Identifiable;
import jakarta.validation.constraints.NotBlank;

import java.time.LocalDate;

public record Laboratory(
        @NotBlank(message = "연구실 ID는 필수입니다.") String labId,
        @NotBlank(message = "연구실명은 필수입니다.") String labName,
        @NotBlank(message = "건물명은 필수입니다.") String buildingName,
        @NotBlank(message = "층은 필수입니다.") String floor,
        @NotBlank(message = "호실은 필수입니다.") String roomNo,
        @NotBlank(message = "소속 학과는 필수입니다.") String departmentName,
        @NotBlank(message = "연구실 책임자 ID는 필수입니다.") String managerId,
        @NotBlank(message = "연구실 책임자명은 필수입니다.") String managerName,
        String contactNo,
        @NotBlank(message = "관리 등급은 필수입니다.") String managementGrade,
        @NotBlank(message = "사용 여부는 필수입니다.") String isActive,
        String createdAt
) implements Identifiable {
    public Laboratory {
        createdAt = createdAt == null || createdAt.isBlank()
                ? LocalDate.now().toString()
                : createdAt;
    }

    @Override
    public String id() {
        return labId;
    }
}

File: src/features/laboratory/domain/Laboratory.js
import { createEntity } from '../../../shared/domain/createEntity';

const schema = {
  labId: { label: '연구실 ID', required: true },
  labName: { label: '연구실명', required: true },
  buildingName: { label: '건물명' },
  floor: { label: '층' },
  roomNo: { label: '호실' },
  departmentName: { label: '소속 부서' },
  managerId: { label: '관리자 ID' },
  managerName: { label: '관리자명' },
  contactNo: { label: '연락처' },
  managementGrade: { label: '관리 등급', required: true },
  isActive: { label: '사용 여부', required: true },
  createdAt: { label: '등록일' },
};

export function createLaboratory(input) {
  return createEntity(
    { ...input, createdAt: input.createdAt || new Date().toISOString().slice(0, 10) },
    schema
  );
}
